modules/network/src/main/java/org/apache/ignite/internal/network/recovery/RecoveryClientHandshakeManager.java

```diff
@@ -136,14 +162,29 @@ public void onMessage(NetworkMessage message) {
             this.remoteLaunchId = msg.launchId();
             this.remoteConsistentId = msg.consistentId();
 
-            this.recoveryDescriptor = recoveryDescriptorProvider.getRecoveryDescriptor(
-                    remoteConsistentId,
-                    remoteLaunchId,
-                    connectionId,
-                    false
-            );
+            channelCreationListener.notifyInboundChannelCreation(this.remoteConsistentId, this.connectionId)
```

Comment:
Maybe all of this should be extracted to `onHandshakeStartMessage` instead of being inside of a single big method